examples/with-typescript-graphql/test/index.test.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import Index from '../pages'
+import renderer from 'react-test-renderer'
+import { MockedProvider } from '@apollo/client/testing'
+import gql from 'graphql-tag'
+
+function timeout(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+const mocks: any = [
+  {
+    request: {
+      query: gql`
+        query Viewer {
+          viewer {
+            ...Partial
+            status
+          }
+        }
+
+        fragment Partial on User {
+          id
+          name
+        }
+      `,
+      variables: {},
+    },
+    result: {
+      data: {
+        viewer: {
+          id: 'Baa',
+          name: 'Baa',
+          status: 'Healthy',
+        },
+      },
+    },
+  },
+]
+
+describe('Index', () => {
+  it('renders the html we want', async () => {
+    const component = renderer.create(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Index />
+      </MockedProvider>
```

Comment:
I'm not very into tests like this one, this can break very easily if the component uses other kinds of context, like the one used by the Next.js Router. i.e if the page uses `useRouter` this will break.

Unit components like this one are okay for single UI components, but overall they aren't very useful.

Reply:
So, while `components/*.tsx`s should be portable, naturally `pages/*.tsx`s tend to be bound by specific contexts. Here, as an Apollo example, it demonstrates how tests of page components would look. Perhaps, other contexts will be required soon in real-world projects [as another example indicates](https://github.com/piglovesyou/next.js/blob/268d3572bc3c95419b43ed24663f917706bcdb2b/examples/with-typescript-eslint-jest/test/testUtils.ts#L9) but I think it's up to users if they keep maintaining or erase the test.

Would you like tests of context-free components, or still think it's a good idea to including [Apollo Testing](https://www.apollographql.com/docs/react/api/react/testing/) example?